guide/src/migration.md

```diff
@@ -5,13 +5,91 @@ For a detailed list of all changes, see [CHANGELOG.md](https://github.com/PyO3/p
 
 ## from 0.11.* to 0.12
 
+### `PyErr` has been reworked
+
+In PyO3 `0.12` the `PyErr` type has been re-implemented to be significantly more compatible with
+the standard Rust error handling ecosystem. Specificially `PyErr` now implements
+`Error + Send + Sync`, which are the standard traits used for error types.
+
+While this has necessitated the removal of a number of APIs, the resulting `PyErr` type should now
+be much more easier to work with. The following sections list the changes in detail and how to
+migrate to the new APIs.
+
+#### `PyErr::new` and `PyErr::from_type` now require `Send + Sync` for their argument
+
+For most uses no change will be needed. If you are trying to construct `PyErr` from a value that is
+not `Send + Sync`, you will need to first create the Python object and then use
+`PyErr::from_instance`.
+
+Similarly, any types which implemented `PyErrArguments` will now need to be `Send + Sync`.
+
+#### `PyErr`'s contents are now private
+
+Making the `PyErr` internals private makes it possible for PyO3 to implement optimizations safely
```

Comment:
I think this is not *perfectly* true: isn't it more of a safety reason considering normalization?
And it would be maybe too detailed for just a migration guide.